legal-api/src/legal_api/models/alternate_name.py

```diff
@@ -111,6 +119,56 @@ def find_by_name(cls, name: str = None):
         alternate_name = cls.query.filter_by(name=name).filter_by(end_date=None).one_or_none()
         return alternate_name
 
+    @property
+    def office_mailing_address(self):
+        """Return the mailing address."""
+        registered_office = (
+            db.session.query(Office)
+            .filter(Office.legal_entity_id == self.id)
+            .filter(Office.office_type == "registeredOffice")
+            .one_or_none()
+        )
+        if registered_office:
+            return registered_office.addresses.filter(Address.address_type == "mailing")
+        elif (
+            business_office := db.session.query(Office)  # SP/GP
+            .filter(Office.legal_entity_id == self.id)
+            .filter(Office.office_type == "businessOffice")
+            .one_or_none()
+        ):
+            return business_office.addresses.filter(Address.address_type == "mailing")
+
+        return (
+            db.session.query(Address)
+            .filter(Address.legal_entity_id == self.id)
+            .filter(Address.address_type == Address.MAILING)
+        )
```

Comment:
i'm not sure about this.  i don't know if we actually save addresses directly to the LE right now. 

Going to check prod data.

Reply:
so it seems like when we dissolve a business right now, it creates the custodian address which differs from how we have saved other office addresses.  As in it populates both the `legal_entity_id` and the `office_id` in the `addresses` table.

For now, I think we should just add the `alternate_name_id` to `addresses` table too.  This way we can just follow the existing pattern for SPs